x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -712,16 +714,43 @@ private static Set<String> subfields(Set<String> names) {
 
     private PhysicalPlan logicalPlanToPhysicalPlan(LogicalPlan optimizedPlan, EsqlQueryRequest request) {
         PhysicalPlan physicalPlan = optimizedPhysicalPlan(optimizedPlan);
+        final QueryBuilder filter = request.filter();
+        final Holder<Boolean> hasScoring = new Holder<>(false);
+
+        if (filter != null) {
+            physicalPlan.forEachDown(EsQueryExec.class, esQueryExec -> {
+                if (hasScoring.get() == false && esQueryExec.hasScoring()) {
+                    hasScoring.set(true);
+                }
+            });
+            if (hasScoring.get() == false) {
+                physicalPlan.forEachDown(FragmentExec.class, fragmentExec -> {
+                    fragmentExec.fragment().forEachDown(EsRelation.class, esRelation -> {
+                        if (esRelation.output()
+                            .stream()
+                            .anyMatch(a -> a instanceof MetadataAttribute && a.name().equals(MetadataAttribute.SCORE))) {
+                            hasScoring.set(true);
+                        }
+                    });
+                });
+            }
+        }
         physicalPlan = physicalPlan.transformUp(FragmentExec.class, f -> {
-            QueryBuilder filter = request.filter();
             if (filter != null) {
                 var fragmentFilter = f.esFilter();
                 // TODO: have an ESFilter and push down to EsQueryExec / EsSource
                 // This is an ugly hack to push the filter parameter to Lucene
                 // TODO: filter integration testing
-                filter = fragmentFilter != null ? boolQuery().filter(fragmentFilter).must(filter) : filter;
-                LOGGER.debug("Fold filter {} to EsQueryExec", filter);
-                f = f.withFilter(filter);
+                QueryBuilder newFilter;
+                if (fragmentFilter != null) {
+                    newFilter = hasScoring.get()
+                        ? boolQuery().filter(fragmentFilter).must(filter)
+                        : boolQuery().filter(fragmentFilter).filter(filter);
```

Comment:
This is an enhancement to existent code where we just take the filter from the request and push it down to ES. Whereas, I believe that if the query doesn't need scoring, we should wrap the filter in a `bool` `filter` statement so that scoring is completely ignored when the query is executed on ES.